### PR TITLE
Refactor OpenAPI spec renderer

### PR DESCRIPTION
**Pull request by tijmenb**
### Context

The current code to generate the OpenAPI spec has a few bugs. They aren't easy to fix because the code is hard to follow without context in some places.

### Changes proposed in this pull request

This PR does too much (sorry about that).

1. Refactor the code to be more structured. It also removes the implementation via a middleman extension in favour of a direct approach.
2. Fixes some of the issues that we've seen (incorrect generation of examples)
3. Improves the layout

This also incorporates changes by @chubberlisk (https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training-tech-docs/pull/72).  

### Guidance to review

- Read the code to see if it makes sense
- Run the site to see if the API docs look good

### Release notes

- [x] [If needed](/README.md#release-notes), the [release notes](/source/release-notes.html.md) have been updated

### Link to Trello card

https://trello.com/c/rjeDYEGQ/996-raise-pr-for-the-tech-docs-openapi-formatter-refactor

### CODE REVIEW

File: lib/govuk_tech_docs/api_reference/renderer.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
+module GovukTechDocs
+  module ApiReference
+    class Renderer
+      def initialize(app, document)
+        @app = app
```
Comment (chubberlisk) [inline lib/govuk_tech_docs/api_reference/renderer.rb]: `@app` doesn't seem to be used anywhere else... 👀 Potentially we can remove it?

Comment (chubberlisk) [reply lib/govuk_tech_docs/api_reference/renderer.rb]: Has since been updated in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training-tech-docs/pull/74/commits/6568e7c0aeb96f6956dc0ac72a07fede907210b3.

**Comment by duncanjbrown** [comment]
:+1: :+1:
